PhotoFeed/app/src/main/java/com/mobileproto/jovanduy/photofeed/HttpHandler.java

```diff
@@ -0,0 +1,80 @@
+package com.mobileproto.jovanduy.photofeed;
+
+
+import android.content.Context;
+import android.telecom.Call;
+import android.util.Log;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+/**
+ * Created by Jordan on 9/24/15.
+ */
+public class HttpHandler {
+
+    public RequestQueue queue;
+
+    public HttpHandler(Context context) { queue = Volley.newRequestQueue(context); }
+
+    /**
+     * sends to GET request to
+     * @param searchQuery String representing what is being searched
+     * @param callback callback
+     */
+    public void searchImages(String searchQuery, int startIndex, final Callback callback) {
+        String query = searchQuery.replaceAll(" ", "+");
+        String url = "https://www.googleapis.com/customsearch/v1";
+        String key = "AIzaSyDiS0H5_uYE54YwVqBKvsz2yPbDcWNQsxw";
+        String cx = "003519602263697178317:rf_putw_noi";
+        url += "?key=" + key;
+        url += "&cx=" + cx;
+        url += "&q=" + query;
+        url += "&searchType=image";
+        url += "&num=9";
+        url += "&start=" + String.valueOf(startIndex);
+
+        JsonObjectRequest request = new JsonObjectRequest(
+                Request.Method.GET,
+                url,
+                new JSONObject(),
+                new Response.Listener<JSONObject>() {
+                    @Override
+                    public void onResponse(JSONObject response) {
```

Comment:
Why don't you modify your callback to take in a JsonObject and then also have your callback do these lines of code since it already does some processing of your response.